metrics/black_list.go

```diff
@@ -0,0 +1,72 @@
+package metrics
+
+import (
+	"regexp"
+	"strings"
+)
+
+// BlackList encapsulates the logic needed to filter based on a string.
+type BlackList struct {
+	list        map[string]struct{}
+	rList       []*regexp.Regexp
+}
+
+// New constructs a new BlackList based on a white- and a
+// blacklist. Only one of them can be not empty.
+func New(b map[string]struct{}) (*BlackList, error) {
+	black := copyList(b)
+	var list map[string]struct{}
+	list = black
+
+	return &BlackList{
+		list:        list,
+	}, nil
+}
+
+// Parse parses and compiles all of the regexes in the BlackList.
+func (l *BlackList) Parse() error {
+	var regexes []*regexp.Regexp
+	for item := range l.list {
+		r, err := regexp.Compile(item)
+		if err != nil {
+			return err
+		}
+		regexes = append(regexes, r)
+	}
+	l.rList = regexes
+	return nil
+}
+
+
+// IsIncluded returns if the given item is included.
+func (l *BlackList) IsIncluded(item string) bool {
```

Comment:
It seems to be the only function with BlackList receiver that is used. We should remove all other unused functions.

Reply:
Yes,your are right,I have deleted all other unused functions.